presto-main/src/main/java/com/facebook/presto/dispatcher/LocalDispatchQuery.java

```diff
@@ -179,7 +179,10 @@ private void waitForMinimumWorkers()
     {
         ListenableFuture<?> minimumWorkerFuture = clusterSizeMonitor.waitForMinimumWorkers();
         // when worker requirement is met, wait for query execution to finish construction and then start the execution
-        addSuccessCallback(minimumWorkerFuture, () -> addSuccessCallback(queryExecutionFuture, this::startExecution));
+        addSuccessCallback(minimumWorkerFuture, () -> {
+            stateMachine.endWaitingForResources();
```

Comment:
thinking out loud, this will lead to a time period which is not accountable anywhere (i.e. T2 - T1). Is it possible to have it counted somewhere else? May be in the dispatch time itself by moving the state machine to dispatching instead?

Reply:
Thank you for your review. I have RWT counted in beginDispatching and use the `endResourceWaitingNanos` to save the T1, which means that T2 - T1 = beginDispatchingNanos - endResourceWaitingNanos.